hoomd/md/force.py

```diff
@@ -381,21 +357,137 @@ def _add(self, simulation):
         super()._add(simulation)
 
     def _attach(self):
+
         # initialize the reflected c++ class
-        if isinstance(self._simulation.device, hoomd.device.CPU):
+        sim = self._simulation
+
+        if isinstance(sim.device, hoomd.device.CPU):
             my_class = _md.ActiveForceCompute
         else:
             my_class = _md.ActiveForceComputeGPU
 
-        self._cpp_obj = my_class(
-            self._simulation.state._cpp_sys_def,
-            self._simulation.state._get_group(self.filter),
-            self.rotation_diff,
-            _hoomd.make_scalar3(0, 0, 0),
-            0,
-            0,
-            0,
+        self._cpp_obj = my_class(sim.state._cpp_sys_def,
+                                 sim.state._get_group(self.filter),
+                                 self.rotation_diff)
+
+        # Attach param_dict and typeparam_dict
+        super()._attach()
+
+
+class ActiveConstraint(Force):
```

Comment:
I'm going to rename this class name to `ActiveOnManifold`. `ActiveConstraint` suggests that it is the constraint itself that is somehow active.